saleor/graphql/checkout/mutations/checkout_delivery_method_update.py

```diff
@@ -266,9 +267,24 @@ def _update_delivery_method(
         invalidate_prices_updated_fields = invalidate_checkout(
             checkout_info, lines, manager, save=False
         )
-        checkout.save(
-            update_fields=checkout_fields_to_update + invalidate_prices_updated_fields
-        )
+        try:
+            checkout.save(
+                update_fields=(
+                    checkout_fields_to_update + invalidate_prices_updated_fields
+                )
+            )
+        except DatabaseError as error:
+            if type(error) is DatabaseError:
```

Comment:
so the best solution that I see is the one where just before `save()` we check if the item still exists. not the perfect one but it should mitigate this issue. I'm not sure capturing `DatabaseError` by its str representation is the proper way to filter errors, while it can be changed.